### PR TITLE
Support custom package in codestarts

### LINKED ISSUE #13362: Generated projects don't follow the given package name (codestarts)
**Describe the bug**
This bug is not affecting the RESTEasy and SpringWeb codestarts

**Expected behavior**
Given package name should be used (instead of `org.acme`) for all examples

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #13362

### CODE REVIEW

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/SmartPackageFileStrategyHandler.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        final Optional<String> optPackageName = NestedMaps.getValue(data, "project.package-name");
+        Path targetPath;
+        String content = codestartFiles.get(0).getContent();
+        if (optPackageName.isPresent()) {
+            final String packageName = optPackageName.get();
+            final String packageNameAsDir = packageName.replace(".", "/");
+            targetPath = targetDirectory.resolve(relativePath.replace("org/acme", packageNameAsDir));
+            content = content
+                .replace("import org.acme", "import " + packageName)
```
Comment (maxandersen) [inline independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/SmartPackageFileStrategyHandler.java:34]: minor thing but I would extract the "org.acme" and "org/acme" into a constant to make it more clear its a special "marker"

**Review by maxandersen — APPROVED** [review]
just minor suggestion - otherwise all good and can be merged.

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/SmartPackageFileStrategyHandler.java (hunk 1 above)
Comment (ia3andy) [reply independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/SmartPackageFileStrategyHandler.java:34]: Sure, this is a draft still :)

**Comment by ia3andy** [comment]
cc @FroMage

**Review by FroMage — APPROVED** [review]

**Comment by ia3andy** [comment]
@glefloch the error I am getting `QuarkusModelBuilderTest` seems unrelated to my changes, any thoughts?

**Comment by glefloch** [comment]
In both actions, this is the `MultiModuleUberJarTest.testUberJarForMultiModule()` that is failing. This test runs a `quarkusBuild` on an existing project, so I don't think this is related to your changes. But that's weird that both gradle CI action failed on the same test.

**Comment by ia3andy** [comment]
> In both actions, this is the `MultiModuleUberJarTest.testUberJarForMultiModule()` that is failing. This test runs a `quarkusBuild` on an existing project, so I don't think this is related to your changes. But that's weird that both gradle CI action failed on the same test.

Yeah, I  found it weird too, that's why I pinged you

**Comment by ia3andy** [comment]
@glefloch the same test is passing locally

**Comment by ia3andy** [comment]
ok it seems to be not only this PR: https://github.com/quarkusio/quarkus/pull/13379

**Comment by ia3andy** [comment]
I created https://github.com/quarkusio/quarkus/issues/13382

**Comment by ia3andy** [comment]
and another unrelated error on Logging-gelf.. 🤯

**Comment by ia3andy** [comment]
@maxandersen too bad it didn't make the 1.10.0.Final by a few hours, it will be in the next fix version.

**Pull request merged**